### PR TITLE
Only add `-DSPIRV_CHECK_CONTEXT` for Debug builds

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

**Comment by dneto0** [comment]
I suggested this change because IREE swiftshader-based tests were failing due to timeouts.

@s-perron has final say.

**Review by dneto0 — COMMENTED** [review]
LGTM but needs @s-perron approval.

**Review by s-perron — APPROVED** [review]

**Pull request merged**
